File: java/src/main/java/de/gdata/vaas/exceptions/VaasServerException.java
package de.gdata.vaas.exceptions;

/**
 * The server encountered an internal error.
 * Recommended actions:
 * * You may retry the request after a certain delay.
 * * If the problem persists contact G DATA.
 */
public class VaasServerException extends Exception {
    public VaasServerException(String message, Exception cause) {
        super(message != null ? message : "Server error", cause);
    }

    public VaasServerException(String message) {
        this(message, null);
    }

    public VaasServerException() {
        this(null, null);
    }
}

File: dotnet/Vaas/src/Vaas/Exceptions/VaasServerException.cs
using System;

namespace Vaas.Exceptions;

/// <summary>The server encountered an internal error.</summary>
/// <remarks>
/// Recommended actions:
/// <ul>
///   <li>You may retry the request after a certain delay.</li>
///   <li>If the problem persists contact G DATA.</li>
/// </ul>
/// </remarks>
public class VaasServerException : Exception
{
    public VaasServerException(string? message)
        : base(message) { }

    public VaasServerException(string? message, Exception? innerException)
        : base(message, innerException) { }
}
